dll/win32/shell32/iconcache.cpp

```diff
@@ -654,34 +654,15 @@ void SIC_Destroy(void)
  */
 static int SIC_LoadOverlayIcon(int icon_idx)
 {
-    WCHAR buffer[1024], wszIdx[8];
-    HKEY hKeyShellIcons;
-    LPCWSTR iconPath;
+    LPWSTR iconPath;
     int iconIdx;
 
     iconPath = swShell32Name;    /* default: load icon from shell32.dll */
     iconIdx = icon_idx;
 
-    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE, L"Software\\Microsoft\\Windows\\CurrentVersion\\Explorer\\Shell Icons",
-                      0, KEY_READ, &hKeyShellIcons) == ERROR_SUCCESS)
+    if (!HLM_GetIconW(icon_idx, iconPath, MAX_PATH, &iconIdx))
```

Comment:
Shouldn't it attempt to read from the Current User registry here first? (I know the original code didn't, but check if Windows does this)

Reply:
Well, it completely depends on the icon which is to be loaded. For virtual namespace folders, it is always HKCU, but for other icons, it's HKLM instead. Since we're currently using this only for shortcut icon, we know exactly that this icon is defined in HKLM.
OK, just checked: on Windows XP SP3 at least, when loading virtual namespace folder icon, it indeed reads from HKCU first, but when reading other icons, e. g. custom shortcut icon, it obviously does not read from HKCU, since it is not defined in that key, so it queries the value 29 to load from HKLM instead. And then, only in case of failure, everything fallbacks to HKCR defaults. Same as it done by me. :slightly_smiling_face: 
So it fully depends on the icon which we want to load using this function. If we actually use it only for shortcut icon, which is defined in HKLM anyway, then it's fine to use `HLM_GetIconW` and in case of failure `HCR_GetIconW`. Now I only need to know which GUID defines stortcut icon. I'm sure default shortcut icon should be always defined in HKCR too.
![VirtualBox_Windows XP_07_02_2024_11_35_56](https://github.com/reactos/reactos/assets/26385117/f1868ce6-87a4-47be-8052-0670645f04e3)

Reply:
Just checked: shortcut icon isn't defined in HKCU, there are only: virtual namespace folders icons and some Start Menu icons. That's all.

Reply:
Ah, as it appears, there is no normal shortcut CLSID defined at all. I can't find it as in registry, same in our code base. So it is always loaded from HKLM, and therefore, the code is OK as-is.